R/query-join.R

```diff
@@ -335,21 +335,18 @@ sql_table_prefix <- function(con, var, table = NULL) {
     cli_abort("{.arg var} must be a bare character.", .internal = TRUE)
   }
   var <- sql_escape_ident(con, var)
-
-  if (!is.null(table)) {
-    table <- as_table_ident(table)
-    table <- escape(table, collapse = NULL, con = con)
-    sql(paste0(table, ".", var))
-  } else {
-    var
-  }
+  sql_table_name_prefix(con, table, var)
 }
 
 sql_star <- function(con, table = NULL) {
-  var <- sql("*")
+  sql_table_name_prefix(con, table, sql("*"))
+}
+
+sql_table_name_prefix <- function(con, table, var) {
```

Comment:
I think `var` should be escaped in this function.

Reply:
Yeah, that makes sense. It required a fix to `sql_escape_ident.TestConnection` which previously escaped everything, even if already SQL.